src/installer/corehost/cli/apphost/bundle/dir_utils.cpp

```diff
@@ -91,3 +97,46 @@ void dir_utils_t::fixup_path_separator(pal::string_t& path)
         }
     }
 }
+
+// Retry the rename operation with some wait in between the attempts.
+// This is an attempt to workaround for possible file locking caused by AV software.
+
+bool dir_utils_t::rename_with_retries(pal::string_t& old_name, pal::string_t& new_name, bool& dir_exists)
+{
+    for (int retry_count=0; retry_count < 500; retry_count++)
+    {
+        if (pal::rename(old_name.c_str(), new_name.c_str()) == 0)
+        {
+            return true;
+        }
+        bool should_retry = errno == EACCES;
```

Comment:
Is `errno` with `EACCES` also used on Windows?

Reply:
Yes.
https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/rename-wrename?view=vs-2019